Add support for notifications, updated documentation, code refactoring

kyletmiller opened the pull request:
This set of commits adds support for macOS notifications when the Location is changed. It also expands the documentation a bit and incorporates some code refactoring.

eprev commented:
Hey Kyle, I appreciate you effort but cannot accept this pull request, sorry.

1. It's changing too many things in a single pull request.
2. I can't accept changes like curly-bracing variable names. What's the purpose of it?
3. I'd be happy to improve the documentation, and not replacing it.
4. Notifications – one line you can put in `~.locations` scripts, so it's probably worth mentioning in the documentation, but I can't see it as a feature.

kyletmiller commented:
Hi Anton:

Thanks for reviewing the changes (and for creating the codebase to begin with)!

No worries about accepting the MR; I'm happy to maintain a fork. I wanted to offer the changes back if you'd like them. Feel free to incorporate (or not incorporate) as you see fit.

To address your points:

1. I feel these commits/changes complement each other in useful ways. I created them as individual commits (as opposed to one giant commit) so they could be cherrypicked, if helpful.
2. Agreed that in and of itself, this likely wouldn't warrant a MR. As part of the refactoring, I incorporated this change for readability and consistency. It's certainly a personal preference and not a technical requirement. Some good dialogue in support of such usage can be found at the following:
* https://stackoverflow.com/questions/8748831/when-do-we-need-curly-braces-around-shell-variables
* http://mywiki.wooledge.org/BashFAQ/082
3. Understood, and apologies for stepping on any toes. I tried to retain the bulk of your already great documentation and augment it with additional information on existing and new features.
4. I found maintaining this across numerous Location scripts to be cumbersome. Since my desired notifications are more than just 'network changed', incorporating this into the main script logic seemed right, and simplified my Location scripts.

Again, thanks for reviewing the code and providing the great app to the community.


Thanks,
kyletmiller

eprev commented:
Thank you for understanding!
